pipeline/ingestor/ingestor_common/src/main/scala/uk/ac/wellcome/platform/ingestor/common/Indexer.scala

```diff
@@ -0,0 +1,81 @@
+package uk.ac.wellcome.platform.ingestor.common
+
+import com.sksamuel.elastic4s.ElasticDsl._
+import com.sksamuel.elastic4s.requests.bulk.{BulkResponse, BulkResponseItem}
+import com.sksamuel.elastic4s.requests.common.VersionType.ExternalGte
+import com.sksamuel.elastic4s.{ElasticClient, Index, Indexable, Response}
+import grizzled.slf4j.Logging
+import uk.ac.wellcome.elasticsearch.model.{CanonicalId, Version}
+
+import scala.concurrent.{ExecutionContext, Future}
+
+trait Indexer[T] extends Logging {
+  val client: ElasticClient
+
+  implicit val ec: ExecutionContext
+  implicit val indexable: Indexable[T]
+  implicit val id: CanonicalId[T]
+  implicit val version: Version[T]
+  val indexName: Index
+
+  final def index(documents: Seq[T]): Future[Either[Seq[T], Seq[T]]] = {
+
+    debug(s"Indexing ${documents.map(d => id.canonicalId(d)).mkString(", ")}")
+
+    val inserts = documents.map { document =>
+      indexInto(indexName.name)
+        .version(version.version(document))
+        .versionType(ExternalGte)
+        .id(id.canonicalId(document))
+        .doc(document)
+    }
+
+    client
+      .execute {
+        bulk(inserts)
+      }
+      .map { response: Response[BulkResponse] =>
+        if (response.isError) {
+          warn(s"Error from Elasticsearch: $response")
```

Comment:
Why `warn` over `error`?

Reply:
Good question. This is copied over from `WorkIndexer` so it's how it currently works and I didn't even see it. Will change, thanks.